### ISSUE TITLE
Cannot checkout repo with symlinks

### ISSUE THREAD

**Issue by drjayvee** [spckio/spck-issues#558]
Our repo has symlinks which are used only for testing and deployment.

It would be nice if I could chose to ignore the symlink (perhaps with the option to create a copy at the symlink location). The app should then ignore any changes to those files. That way, I could at least checkout and work on the repo.

Context: I'm trying to set up my tablet as an emergency workstation to commit hotfixes for when I'm out of the office.

**Comment by spck-io** [comment]
This is a slightly complicated issue as ignoring the symlinks will cause an issue with commit hashes not matching the remote. Modifying the contents of the file will also cause the git hashes to not match with the remote which can cause all sorts of problems, as well as slow down git operations a lot by needing to read the contents of the file to check if it's a symlink instead of just reading the file object metadata. There is nothing in the code explicitly preventing symlinks from being written, it attempts it, but is most likely blocked by the Android OS. I think it's possible to root your phone to bypass the SELinux policy on Android, but I wouldn't recommend it.

**Comment by drjayvee** [comment]
Thanks for the detailed explanation.

Rooting Android to be able to push code definitely seems like a very bad idea security-wise.

I guess this is a won't fix then.
